lib/services/services.go

```diff
@@ -44,36 +43,40 @@ func NewServiceRegistry() *ServiceRegistry {
 
 // RegisterService stores a new service in the map. If a service of that type has been seen
 func (s *ServiceRegistry) RegisterService(service Service) {
+	if service == nil {
+		log.Warn("Could not be able to register a nil service")
+		return
+	}
+
 	kind := reflect.TypeOf(service)
 	if _, exists := s.services[kind]; exists {
 		log.Warn("Tried to add service type that has already been seen", "type", kind)
 		return
 	}
 	s.services[kind] = service
-	s.serviceTypes = append(s.serviceTypes, kind)
```

Comment:
I think the problem with this change is that previously the services were being started and stopped in a deterministic order, but now when iterating over the map it's non-deterministic. this might mess up the start/stop logic, since some services need to be started/stopped before others, can you confirm starting and stopping the node still works without errors when changing this?

Reply:
I've started the node and sometime later stoped with **ctrl+c**

> ./bin/gossamer --chain dev
> ... waiting some time
> ^CINFO[06-11|12:42:46] signal interrupt, shutting down...       pkg=dot caller=node.go:432
> INFO[06-11|12:42:46] Stopping services...                     caller=services.go:74

I think the node stoped without errors but I just have rolled back the code to avoid unexpected node failures

Reply:
nice, I'm not 100% sure if it would cause errors if it's non deterministic but I remember previously it had to be in a certain order. maybe we can revisit this later to see if it's still an issue